Chore/add identity 1.3 (#1555)
iotaledger/iota-wiki: docs/build/identity.rs/1.3/docs/references/specifications/revocation-timeframe-2024.mdx

---
title: Revocation Timeframe 2024
sidebar_label: Revocation Timeframe 2024

> description: The specification for an embeddedable time-based revocation method - `RevocationTimeframe2024`.

Reviewer: ```suggestion
description: The specification for an embeddable time-based revocation method - `RevocationTimeframe2024`.
```

image: /img/Identity_icon.png
tags:
  - DID
  - specs
  - specifications
  - revocation
  - bitmap
---

# Revocation Timeframe 2024

## Abstract

This specification describes a new revocation mechanism - `RevocationTimeframe2024` - that extends [`RevocationBitmap2022`](../revocation-bitmap-2022)

> in order to address its linkability security concerns, at least when used in combination with selectively-disclosable (SD-able) VCs.

Reviewer: ```suggestion
to address its linkability security concerns, at least when used in combination with selectively disclosable (SD-able) VCs.
```

## Introduction

`RevocationBitmap2022` allows for a simple and straightforward way for an issuer to invalidate an issued VC before its expiration. While this method prevents the analysis of usage by the issuer through hosting the revocation information on-chain

> it comes with a high risk of linkability by colluding verfiers as they can store the bitmap index and use it to identify users. Additionally verifiers can monitor the revocation information and at later point, outside of nay interaction, check the revocation status.

Reviewer: ```suggestion
it comes with a high risk of linkability by colluding verifiers as they can store the bitmap index and use it to identify users. Additionally, verifiers can monitor the revocation information and at a later point, outside of any interaction, check the revocation status.
```

To address this privacy concern, `RevocationTimeframe2024` was designed as an extension that builds on top of `RevocationBitmap2022`, therefore sharing
most of its logic.

## Concepts
`RevocationTimeframe2024` should be used together with selectively disclosable credentials - either [SD-JWT](../../../how-tos/verifiable-credentials/selective-disclosure)
or [ZK-SD](../../../how-tos/verifiable-credentials/zero-knowledge-selective-disclosure) - in order to conceal to the verifier

> the credential's index in the issuers revocation bitmap to avoid linkability.

Reviewer: ```suggestion
the credential's index in the issuer's revocation bitmap to avoid linkability.
```

> ### Validity Timeframe

Reviewer: ```suggestion
### Validity Timeframe

```

> If the revocation index is concealed to the verifier how can it assert the validity of the presented credential?

Reviewer: ```suggestion
If the revocation index is concealed from the verifier how can it assert the validity of the presented credential?
```

> To solve this issue `RevocationTimeframe2024` introduces the concept of a _validity timeframe_, i.e. a limited span of time

Reviewer: ```suggestion
To solve this issue `RevocationTimeframe2024` introduces the concept of a _validity timeframe_, i.e. a limited time span
```

> in which the credential is guaranted to be non-revoked. By having a validity timeframe embedded in the credentials status

Reviewer: ```suggestion
in which the credential is guaranteed to be non-revoked. By having a validity timeframe embedded in the credential's status
```

> the verifier can prove the credentials validity by verifying the credentials signature.

Reviewer: ```suggestion
the verifier can prove the credential's validity by verifying the credential's signature.
```

> The downside of this mechanism is that the holder of a credential using `RevocationTimeframe2024` has to contact the credentials issuer

Reviewer: ```suggestion
The downside of this mechanism is that the  credential holder using `RevocationTimeframe2024` has to contact the credential's issuer
```

to update the signature - i.e. re-issue the credential - at the end of the credentials validity timeframe. Furthermore,

> given as how a credentials validity timeframe proves the validity of the credential itself, the updates made by the issuer to the credentials status -

Reviewer: ```suggestion
given how a credentials validity timeframe proves the validity of the credential itself, the updates made by the issuer to the credential's status -
```

> i.e revoking or unrevoking it - won't be reflected on the credential until the start of a new validity timeframe. For this reason,

Reviewer: ```suggestion
i.e., revoking or un-revoking it - won't be reflected on the credential until a new validity timeframe starts. For this reason,
```

issuers should choose a validity timeframe length with respect to how frequently they expect to change the credential's status;

> frequent updates deems shorter validity timeframes.

Reviewer: ```suggestion
frequent updates deem shorter validity timeframes.
```

:::note
A credential holder does not need to have its credential re-issued at the end of every validity timeframe, but only when

> they need to present the credential to a verifier and the credentials validity timeframe has expired.

Reviewer: ```suggestion
they need to present the credential to a verifier and its validity timeframe has expired.
```

| :---------------------- | :--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| `id`                    | The constraints on the `id` property are listed in the [Verifiable Credentials Data Model specification](https://www.w3.org/TR/vc-data-model/). The `id` MUST be a [DID URL](https://www.w3.org/TR/did-core/#did-url-syntax) that is the URL to a [Revocation Bitmap Service](#revocation-bitmap-service) in the DID Document of the issuer. |

> | `type`                  | The `type` property MUST be `"RevocationBitmap2022"`.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                          |

Reviewer 2: ```suggestion
| `type`                  | The `type` property MUST be `"RevocationTimeframe2024"`.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                          |
```

> | `revocationBitmapIndex` | The `revocationBitmapIndex` property MUST be an unsigned, 32-bit integer expressed as a string. This is the index of the credential in the issuers revocation bitmap. Each index SHOULD be unique among all credentials linking to the same [Revocation Bitmap Service](#revocation-bitmap-service).                                                                                                                                                                                                                                                                                                                          |

Reviewer 3: I would add a sentence at the end like this: 

""This value is _not disclosed_ to the verifier by default"."

> | `revocationBitmapIndex` | The `revocationBitmapIndex` property MUST be an unsigned, 32-bit integer expressed as a string. This is the index of the credential in the issuers revocation bitmap. Each index SHOULD be unique among all credentials linking to the same [Revocation Bitmap Service](#revocation-bitmap-service).                                                                                                                                                                                                                                                                                                                          |

Reviewer 3: ```suggestion
| `revocationBitmapIndex` | The `revocationBitmapIndex` property MUST be an unsigned, 32-bit integer expressed as a string. This is the index of the credential in the issuers revocation bitmap. Each index SHOULD be unique among all credentials linking to the same [Revocation Bitmap Service](#revocation-bitmap-service). To ensure user unlinkability, this value MUST NOT be disclosed to the verifier (this is done by default).                                                                                                                                                                                                                                                                                                                         |
```

| `startValidityTimeframe`| The `startValidityTimeframe` property MUST be a [RFC 3339](https://datatracker.ietf.org/doc/html/rfc3339)-compliant timestamp. |
| `endValidityTimeframe`| The `endValidityTimeframe` property MUST be a [RFC 3339](https://datatracker.ietf.org/doc/html/rfc3339)-compliant timestamp. |

#### Example

An example of a verifiable credential with a `credentialStatus` of type `RevocationTimeframe2024`.

```json
{
  "@context": "https://www.w3.org/2018/credentials/v1",
  "id": "https://example.edu/credentials/3732",
  "type": ["VerifiableCredential", "UniversityDegreeCredential"],
  "credentialSubject": {
    "id": "did:iota:B8DucnzULJ9E8cmaReYoePU2b7UKE9WKxyEVov8tQA7H",
    "GPA": "4.0",
    "degree": "Bachelor of Science and Arts",
    "name": "Alice"
  },
  "issuer": "did:iota:EvaQhPXXsJsGgxSXGhZGMCvTt63KuAFtaGThx6a5nSpw",
  "issuanceDate": "2022-06-13T08:04:36Z",
  "credentialStatus": {

>     "id": "did:iota:EvaQhPXXsJsGgxSXGhZGMCvTt63KuAFtaGThx6a5nSpw?index=5#revocation",

Reviewer 3: ```suggestion
    "id": "did:iota:EvaQhPXXsJsGgxSXGhZGMCvTt63KuAFtaGThx6a5nSpw#my-revocation-service",
```

Reviewer 3: The change reflects what has been done in the `10_zkp_revocation` example. Note that the index MUST not be disclosed.

---
title: Revocation Timeframe 2024
sidebar_label: Revocation Timeframe 2024
description: The specification for an embeddedable time-based revocation method - `RevocationTimeframe2024`.
image: /img/Identity_icon.png
tags:
  - DID
  - specs
  - specifications
  - revocation
  - bitmap
---

# Revocation Timeframe 2024

## Abstract

This specification describes a new revocation mechanism - `RevocationTimeframe2024` - that extends [`RevocationBitmap2022`](../revocation-bitmap-2022)

> in order to address it's linkability security concerns, at least when used in combination with selectively-discolable (SD-able) VCs.

Reviewer 4: ```suggestion
in order to address its linkability security concerns, at least when used in combination with selectively-disclosable (SD-able) VCs.
```

## Introduction

> `RevocationBitmap2022` allows for a simple and straightforward way for an issuer to invalidate an issued VC before it's expiration. This mechanism though

Reviewer 4: (in addition to the previous suggestion)

```suggestion
`RevocationBitmap2022` allows for a simple and straightforward way for an issuer to invalidate an issued VC before its expiration. This mechanism though
```

> comes with a high risk of linkability, which is especially relavant for bitmaps stored on-chain - where the storage space limits the maximum size for

Author: ```suggestion
it comes with a high risk of linkability by colluding verfiers as they can store the bitmap index and use it to identify users. Additionally verifiers can monitor the revocation information and at later point, outside of nay interaction, check the revocation status.
```

> the revocation bitmap, reducing herd privacy.

Author: ```suggestion
the revocation bitmap, reducing herd privacy.
```
```suggestion
```

To address this privacy concern, `RevocationTimeframe2024` was designed as an extension that builds on top of `RevocationBitmap2022`, therefore sharing
most of its logic.

## Concepts
`RevocationTimeframe2024` should be used together with selectively disclosable credentials - either [SD-JWT](../../../how-tos/verifiable-credentials/selective-disclosure)
or [ZK-SD](../../../how-tos/verifiable-credentials/zero-knowledge-selective-disclosure) - in order to conceal to the verifier

> the credential's index in the issuer's revocation bitmap to avoid linkability.

Reviewer 4: (in addition to the previous suggestion)

```suggestion
the credential's index in the issuers revocation bitmap to avoid linkability.
```
